lib/attr_extras/utils.rb

```diff
@@ -1,5 +1,14 @@
 module AttrExtras::Utils
   def self.flat_names(names)
-    names.flatten.map { |x| x.to_s.sub(/!\z/, "") }
+    names.map do |x|
```

Comment:
You could use `flat_map` instead of map + flatten. Mind changing from do/end to curlies [for Jim Weirich](https://github.com/barsoom/devbook/tree/master/styleguide#use-weirich-style-for-blocks)? :) It's possible we haven't been consistent with that throughout this project.